core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -326,36 +390,63 @@ trait RawlsBillingProjectComponent {
         message
       )
     }
+
+    def getLastChange(billingProject: RawlsBillingProjectName): ReadAction[Option[BillingAccountChange]] = {
+      BillingAccountChanges
+        .withProjectName(billingProject)
+        .sortBy(_.id.desc)
+        .result
+        .map(_.headOption)
+    }
   }
 
-  object billingAccountChangeQuery extends TableQuery(new BillingAccountChanges(_)) {
+  implicit class BillingAccountChangeExtensions(query: BillingAccountChangeQuery) {
+
     def create(billingProjectName: RawlsBillingProjectName,
                previousBillingAccount: Option[RawlsBillingAccountName],
                newBillingAccount: Option[RawlsBillingAccountName],
-               userSubjectId: RawlsUserSubjectId): ReadWriteAction[Unit] =
-      billingAccountChangeQuery
+               userSubjectId: RawlsUserSubjectId): ReadWriteAction[Int] =
+      query
         .map(change => (change.billingProjectName, change.previousBillingAccount, change.newBillingAccount, change.userId))
         .insert((billingProjectName.value, previousBillingAccount.map(_.value), newBillingAccount.map(_.value), userSubjectId.value))
-        .ignore
 
-    def lastChange(billingProjectName: RawlsBillingProjectName)
-                  (implicit executionContext: ExecutionContext): ReadWriteAction[Option[BillingAccountChange]] =
-      billingAccountChangeQuery
-        .filter(_.billingProjectName === billingProjectName.value)
-        .sortBy(_.id.desc)
-        .take(1)
-        .result
-        .map(_.headOption)
+    // filters
+    def withId(id: Long): BillingAccountChangeQuery =
+      query.filter(_.id === id)
+
+    def withProjectName(billingProjectName: RawlsBillingProjectName): BillingAccountChangeQuery =
+      query.filter(_.billingProjectName === billingProjectName.value)
+
+    /* SELECT *
+     * FROM BILLING_ACCOUNT_CHANGES BAC,
+     * (  SELECT BILLING_PROJECT_NAME, MAX(ID) AS MAXID
```

Comment:
Was wondering if there's a timestamp column we could use instead of `max(id)` (since this seems to care about most recent records).

Reply:
How would we use the timestamp other than max?

Reply:
Yeah I mean you'd do max(timestamp) as well… was just thinking if there's a situation where max(id) doesn't return the latest record (by timestamp) but I guess not. This seems fine